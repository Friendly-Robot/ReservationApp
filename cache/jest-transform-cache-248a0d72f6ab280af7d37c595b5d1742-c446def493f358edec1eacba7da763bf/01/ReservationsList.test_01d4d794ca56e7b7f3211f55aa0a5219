9a460907e6f6c89e8466267b232954be
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

require("react-native");

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _jestMock = _interopRequireDefault(require("jest-mock"));

var _ReservationsList = _interopRequireDefault(require("../ReservationsList"));

var _reactTestRenderer = _interopRequireDefault(require("react-test-renderer"));

describe('<ReservationsList />', function () {
  var wrapper;
  var props;
  beforeEach(function () {
    props = {
      hotels: [{
        hotelName: 'test',
        __typename: 'Reservation',
        id: '123456',
        name: 'Andy',
        arrivalDate: '04/22/2020',
        departureDate: '04/22/9999'
      }],
      setShowScheduler: _jestMock.default.fn()
    };
    wrapper = (0, _enzyme.shallow)(_react.default.createElement(_ReservationsList.default, null));
  });
  it('should be defined', function () {
    expect(_ReservationsList.default).toBeDefined();
  });
  it('renders correctly', function () {
    var tree = _reactTestRenderer.default.create(_react.default.createElement(_ReservationsList.default, null)).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('should have access to setShowScheduler() function', function () {
    console.log(wrapper.props.setShowScheduler);
  });
});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlJlc2VydmF0aW9uc0xpc3QudGVzdC5qcyJdLCJuYW1lcyI6WyJkZXNjcmliZSIsIndyYXBwZXIiLCJwcm9wcyIsImJlZm9yZUVhY2giLCJob3RlbHMiLCJob3RlbE5hbWUiLCJfX3R5cGVuYW1lIiwiaWQiLCJuYW1lIiwiYXJyaXZhbERhdGUiLCJkZXBhcnR1cmVEYXRlIiwic2V0U2hvd1NjaGVkdWxlciIsImplc3QiLCJmbiIsIml0IiwiZXhwZWN0IiwiUmVzZXJ2YXRpb25zTGlzdCIsInRvQmVEZWZpbmVkIiwidHJlZSIsInJlbmRlcmVyIiwiY3JlYXRlIiwidG9KU09OIiwidG9NYXRjaFNuYXBzaG90IiwiY29uc29sZSIsImxvZyJdLCJtYXBwaW5ncyI6Ijs7QUFJQTs7QUFDQTs7QUFDQTs7QUFDQTs7QUFFQTs7QUFHQTs7QUFFQUEsUUFBUSxDQUFDLHNCQUFELEVBQXlCLFlBQU07QUFDckMsTUFBSUMsT0FBSjtBQUNBLE1BQUlDLEtBQUo7QUFFQUMsRUFBQUEsVUFBVSxDQUFDLFlBQU07QUFDZkQsSUFBQUEsS0FBSyxHQUFHO0FBQ05FLE1BQUFBLE1BQU0sRUFBRSxDQUFDO0FBQ1BDLFFBQUFBLFNBQVMsRUFBRSxNQURKO0FBRVBDLFFBQUFBLFVBQVUsRUFBRSxhQUZMO0FBR1BDLFFBQUFBLEVBQUUsRUFBRSxRQUhHO0FBSVBDLFFBQUFBLElBQUksRUFBRSxNQUpDO0FBS1BDLFFBQUFBLFdBQVcsRUFBRSxZQUxOO0FBTVBDLFFBQUFBLGFBQWEsRUFBRTtBQU5SLE9BQUQsQ0FERjtBQVNOQyxNQUFBQSxnQkFBZ0IsRUFBRUMsa0JBQUtDLEVBQUw7QUFUWixLQUFSO0FBWUFaLElBQUFBLE9BQU8sR0FBRyxxQkFBUSw2QkFBQyx5QkFBRCxPQUFSLENBQVY7QUFDRCxHQWRTLENBQVY7QUFnQkFhLEVBQUFBLEVBQUUsQ0FBQyxtQkFBRCxFQUFzQixZQUFNO0FBQzVCQyxJQUFBQSxNQUFNLENBQUNDLHlCQUFELENBQU4sQ0FBeUJDLFdBQXpCO0FBQ0QsR0FGQyxDQUFGO0FBSUFILEVBQUFBLEVBQUUsQ0FBQyxtQkFBRCxFQUFzQixZQUFNO0FBQzVCLFFBQU1JLElBQUksR0FBR0MsMkJBQVNDLE1BQVQsQ0FBZ0IsNkJBQUMseUJBQUQsT0FBaEIsRUFBc0NDLE1BQXRDLEVBQWI7O0FBQ0FOLElBQUFBLE1BQU0sQ0FBQ0csSUFBRCxDQUFOLENBQWFJLGVBQWI7QUFDRCxHQUhDLENBQUY7QUFLQVIsRUFBQUEsRUFBRSxDQUFDLG1EQUFELEVBQXNELFlBQU07QUFDNURTLElBQUFBLE9BQU8sQ0FBQ0MsR0FBUixDQUFZdkIsT0FBTyxDQUFDQyxLQUFSLENBQWNTLGdCQUExQjtBQUNELEdBRkMsQ0FBRjtBQUdELENBaENPLENBQVIiLCJzb3VyY2VzQ29udGVudCI6WyIvKipcbiAqIEBmb3JtYXRcbiAqL1xuXG5pbXBvcnQgJ3JlYWN0LW5hdGl2ZSc7XG5pbXBvcnQgUmVhY3QgZnJvbSAncmVhY3QnO1xuaW1wb3J0IHsgc2hhbGxvdyB9IGZyb20gJ2VuenltZSc7XG5pbXBvcnQgamVzdCBmcm9tICdqZXN0LW1vY2snO1xuXG5pbXBvcnQgUmVzZXJ2YXRpb25zTGlzdCBmcm9tICcuLi9SZXNlcnZhdGlvbnNMaXN0JztcblxuLy8gTm90ZTogdGVzdCByZW5kZXJlciBtdXN0IGJlIHJlcXVpcmVkIGFmdGVyIHJlYWN0LW5hdGl2ZS5cbmltcG9ydCByZW5kZXJlciBmcm9tICdyZWFjdC10ZXN0LXJlbmRlcmVyJztcblxuZGVzY3JpYmUoJzxSZXNlcnZhdGlvbnNMaXN0IC8+JywgKCkgPT4ge1xuICBsZXQgd3JhcHBlcjtcbiAgbGV0IHByb3BzO1xuXG4gIGJlZm9yZUVhY2goKCkgPT4ge1xuICAgIHByb3BzID0ge1xuICAgICAgaG90ZWxzOiBbe1xuICAgICAgICBob3RlbE5hbWU6ICd0ZXN0JyxcbiAgICAgICAgX190eXBlbmFtZTogJ1Jlc2VydmF0aW9uJyxcbiAgICAgICAgaWQ6ICcxMjM0NTYnLFxuICAgICAgICBuYW1lOiAnQW5keScsXG4gICAgICAgIGFycml2YWxEYXRlOiAnMDQvMjIvMjAyMCcsXG4gICAgICAgIGRlcGFydHVyZURhdGU6ICcwNC8yMi85OTk5JyxcbiAgICAgIH1dLFxuICAgICAgc2V0U2hvd1NjaGVkdWxlcjogamVzdC5mbigpLFxuICAgIH07XG5cbiAgICB3cmFwcGVyID0gc2hhbGxvdyg8UmVzZXJ2YXRpb25zTGlzdCAvPik7XG4gIH0pO1xuXG4gIGl0KCdzaG91bGQgYmUgZGVmaW5lZCcsICgpID0+IHtcbiAgICBleHBlY3QoUmVzZXJ2YXRpb25zTGlzdCkudG9CZURlZmluZWQoKTtcbiAgfSk7XG5cbiAgaXQoJ3JlbmRlcnMgY29ycmVjdGx5JywgKCkgPT4ge1xuICAgIGNvbnN0IHRyZWUgPSByZW5kZXJlci5jcmVhdGUoPFJlc2VydmF0aW9uc0xpc3QgLz4pLnRvSlNPTigpO1xuICAgIGV4cGVjdCh0cmVlKS50b01hdGNoU25hcHNob3QoKTtcbiAgfSk7XG5cbiAgaXQoJ3Nob3VsZCBoYXZlIGFjY2VzcyB0byBzZXRTaG93U2NoZWR1bGVyKCkgZnVuY3Rpb24nLCAoKSA9PiB7XG4gICAgY29uc29sZS5sb2cod3JhcHBlci5wcm9wcy5zZXRTaG93U2NoZWR1bGVyKTtcbiAgfSk7XG59KVxuIl19