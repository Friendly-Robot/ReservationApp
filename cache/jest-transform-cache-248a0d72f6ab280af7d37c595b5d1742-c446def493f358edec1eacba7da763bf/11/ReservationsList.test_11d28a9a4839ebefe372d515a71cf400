7d8c35d91f75f5b64596a79b0c779e37
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

require("react-native");

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _jestMock = _interopRequireDefault(require("jest-mock"));

var _ReservationsList = _interopRequireDefault(require("../ReservationsList"));

var _ReservationsListItem = _interopRequireDefault(require("../ReservationsListItem"));

var _ButtonActionRound = _interopRequireDefault(require("../../../common/ButtonActionRound"));

var _reactTestRenderer = _interopRequireDefault(require("react-test-renderer"));

describe('<ReservationsList />', function () {
  var wrapper;
  var props;
  beforeEach(function () {
    props = {
      hotels: [{
        hotelName: 'test',
        __typename: 'Reservation',
        id: '123456',
        name: 'Andy',
        arrivalDate: '04/22/2020',
        departureDate: '04/22/9999'
      }],
      setShowScheduler: _jestMock.default.fn()
    };
    wrapper = (0, _enzyme.shallow)(_react.default.createElement(_ReservationsList.default, null));
  });
  it('should be defined', function () {
    expect(_ReservationsList.default).toBeDefined();
  });
  it('renders correctly', function () {
    var tree = _reactTestRenderer.default.create(_react.default.createElement(_ReservationsList.default, null)).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('should have access to setShowScheduler() function', function () {
    console.log(JSON.stringify(wrapper));
    var setShowScheduler = wrapper.props()['children'][1].props.onPress;
    setShowScheduler();
    expect(setShowScheduler).toBeCalled;
    console.log(wrapper.find(_ButtonActionRound.default));
  });
});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlJlc2VydmF0aW9uc0xpc3QudGVzdC5qcyJdLCJuYW1lcyI6WyJkZXNjcmliZSIsIndyYXBwZXIiLCJwcm9wcyIsImJlZm9yZUVhY2giLCJob3RlbHMiLCJob3RlbE5hbWUiLCJfX3R5cGVuYW1lIiwiaWQiLCJuYW1lIiwiYXJyaXZhbERhdGUiLCJkZXBhcnR1cmVEYXRlIiwic2V0U2hvd1NjaGVkdWxlciIsImplc3QiLCJmbiIsIml0IiwiZXhwZWN0IiwiUmVzZXJ2YXRpb25zTGlzdCIsInRvQmVEZWZpbmVkIiwidHJlZSIsInJlbmRlcmVyIiwiY3JlYXRlIiwidG9KU09OIiwidG9NYXRjaFNuYXBzaG90IiwiY29uc29sZSIsImxvZyIsIkpTT04iLCJzdHJpbmdpZnkiLCJvblByZXNzIiwidG9CZUNhbGxlZCIsImZpbmQiLCJCdXR0b25BY3Rpb24iXSwibWFwcGluZ3MiOiI7O0FBSUE7O0FBQ0E7O0FBQ0E7O0FBQ0E7O0FBRUE7O0FBQ0E7O0FBQ0E7O0FBR0E7O0FBRUFBLFFBQVEsQ0FBQyxzQkFBRCxFQUF5QixZQUFNO0FBQ3JDLE1BQUlDLE9BQUo7QUFDQSxNQUFJQyxLQUFKO0FBRUFDLEVBQUFBLFVBQVUsQ0FBQyxZQUFNO0FBQ2ZELElBQUFBLEtBQUssR0FBRztBQUNORSxNQUFBQSxNQUFNLEVBQUUsQ0FBQztBQUNQQyxRQUFBQSxTQUFTLEVBQUUsTUFESjtBQUVQQyxRQUFBQSxVQUFVLEVBQUUsYUFGTDtBQUdQQyxRQUFBQSxFQUFFLEVBQUUsUUFIRztBQUlQQyxRQUFBQSxJQUFJLEVBQUUsTUFKQztBQUtQQyxRQUFBQSxXQUFXLEVBQUUsWUFMTjtBQU1QQyxRQUFBQSxhQUFhLEVBQUU7QUFOUixPQUFELENBREY7QUFTTkMsTUFBQUEsZ0JBQWdCLEVBQUVDLGtCQUFLQyxFQUFMO0FBVFosS0FBUjtBQVlBWixJQUFBQSxPQUFPLEdBQUcscUJBQVEsNkJBQUMseUJBQUQsT0FBUixDQUFWO0FBQ0QsR0FkUyxDQUFWO0FBZ0JBYSxFQUFBQSxFQUFFLENBQUMsbUJBQUQsRUFBc0IsWUFBTTtBQUM1QkMsSUFBQUEsTUFBTSxDQUFDQyx5QkFBRCxDQUFOLENBQXlCQyxXQUF6QjtBQUNELEdBRkMsQ0FBRjtBQUlBSCxFQUFBQSxFQUFFLENBQUMsbUJBQUQsRUFBc0IsWUFBTTtBQUM1QixRQUFNSSxJQUFJLEdBQUdDLDJCQUFTQyxNQUFULENBQWdCLDZCQUFDLHlCQUFELE9BQWhCLEVBQXNDQyxNQUF0QyxFQUFiOztBQUNBTixJQUFBQSxNQUFNLENBQUNHLElBQUQsQ0FBTixDQUFhSSxlQUFiO0FBQ0QsR0FIQyxDQUFGO0FBU0FSLEVBQUFBLEVBQUUsQ0FBQyxtREFBRCxFQUFzRCxZQUFNO0FBQzVEUyxJQUFBQSxPQUFPLENBQUNDLEdBQVIsQ0FBWUMsSUFBSSxDQUFDQyxTQUFMLENBQWV6QixPQUFmLENBQVo7QUFDQSxRQUFNVSxnQkFBZ0IsR0FBR1YsT0FBTyxDQUFDQyxLQUFSLEdBQWdCLFVBQWhCLEVBQTRCLENBQTVCLEVBQStCQSxLQUEvQixDQUFxQ3lCLE9BQTlEO0FBQ0FoQixJQUFBQSxnQkFBZ0I7QUFDaEJJLElBQUFBLE1BQU0sQ0FBQ0osZ0JBQUQsQ0FBTixDQUF5QmlCLFVBQXpCO0FBQ0FMLElBQUFBLE9BQU8sQ0FBQ0MsR0FBUixDQUFZdkIsT0FBTyxDQUFDNEIsSUFBUixDQUFhQywwQkFBYixDQUFaO0FBQ0QsR0FOQyxDQUFGO0FBT0QsQ0F4Q08sQ0FBUiIsInNvdXJjZXNDb250ZW50IjpbIi8qKlxuICogQGZvcm1hdFxuICovXG5cbmltcG9ydCAncmVhY3QtbmF0aXZlJztcbmltcG9ydCBSZWFjdCBmcm9tICdyZWFjdCc7XG5pbXBvcnQgeyBzaGFsbG93IH0gZnJvbSAnZW56eW1lJztcbmltcG9ydCBqZXN0IGZyb20gJ2plc3QtbW9jayc7XG5cbmltcG9ydCBSZXNlcnZhdGlvbnNMaXN0IGZyb20gJy4uL1Jlc2VydmF0aW9uc0xpc3QnO1xuaW1wb3J0IFJlc2VydmF0aW9uc0xpc3RJdGVtIGZyb20gJy4uL1Jlc2VydmF0aW9uc0xpc3RJdGVtJztcbmltcG9ydCBCdXR0b25BY3Rpb24gZnJvbSAnLi4vLi4vLi4vY29tbW9uL0J1dHRvbkFjdGlvblJvdW5kJztcblxuLy8gTm90ZTogdGVzdCByZW5kZXJlciBtdXN0IGJlIHJlcXVpcmVkIGFmdGVyIHJlYWN0LW5hdGl2ZS5cbmltcG9ydCByZW5kZXJlciBmcm9tICdyZWFjdC10ZXN0LXJlbmRlcmVyJztcblxuZGVzY3JpYmUoJzxSZXNlcnZhdGlvbnNMaXN0IC8+JywgKCkgPT4ge1xuICBsZXQgd3JhcHBlcjtcbiAgbGV0IHByb3BzO1xuXG4gIGJlZm9yZUVhY2goKCkgPT4ge1xuICAgIHByb3BzID0ge1xuICAgICAgaG90ZWxzOiBbe1xuICAgICAgICBob3RlbE5hbWU6ICd0ZXN0JyxcbiAgICAgICAgX190eXBlbmFtZTogJ1Jlc2VydmF0aW9uJyxcbiAgICAgICAgaWQ6ICcxMjM0NTYnLFxuICAgICAgICBuYW1lOiAnQW5keScsXG4gICAgICAgIGFycml2YWxEYXRlOiAnMDQvMjIvMjAyMCcsXG4gICAgICAgIGRlcGFydHVyZURhdGU6ICcwNC8yMi85OTk5JyxcbiAgICAgIH1dLFxuICAgICAgc2V0U2hvd1NjaGVkdWxlcjogamVzdC5mbigpLFxuICAgIH07XG5cbiAgICB3cmFwcGVyID0gc2hhbGxvdyg8UmVzZXJ2YXRpb25zTGlzdCAvPik7XG4gIH0pO1xuXG4gIGl0KCdzaG91bGQgYmUgZGVmaW5lZCcsICgpID0+IHtcbiAgICBleHBlY3QoUmVzZXJ2YXRpb25zTGlzdCkudG9CZURlZmluZWQoKTtcbiAgfSk7XG5cbiAgaXQoJ3JlbmRlcnMgY29ycmVjdGx5JywgKCkgPT4ge1xuICAgIGNvbnN0IHRyZWUgPSByZW5kZXJlci5jcmVhdGUoPFJlc2VydmF0aW9uc0xpc3QgLz4pLnRvSlNPTigpO1xuICAgIGV4cGVjdCh0cmVlKS50b01hdGNoU25hcHNob3QoKTtcbiAgfSk7XG5cbiAgLy8gaXQoJ3JlbmRlcnMgUmVzZXJ2YXRpb25zTGlzdEl0ZW0nLCAoKSA9PiB7XG4gIC8vICAgZXhwZWN0KHdyYXBwZXIuZmluZChSZXNlcnZhdGlvbnNMaXN0aXRlbSkpLnRvQmVEZWZpbmVkKCk7XG4gIC8vIH0pXG5cbiAgaXQoJ3Nob3VsZCBoYXZlIGFjY2VzcyB0byBzZXRTaG93U2NoZWR1bGVyKCkgZnVuY3Rpb24nLCAoKSA9PiB7XG4gICAgY29uc29sZS5sb2coSlNPTi5zdHJpbmdpZnkod3JhcHBlcikpO1xuICAgIGNvbnN0IHNldFNob3dTY2hlZHVsZXIgPSB3cmFwcGVyLnByb3BzKClbJ2NoaWxkcmVuJ11bMV0ucHJvcHMub25QcmVzcztcbiAgICBzZXRTaG93U2NoZWR1bGVyKCk7XG4gICAgZXhwZWN0KHNldFNob3dTY2hlZHVsZXIpLnRvQmVDYWxsZWRcbiAgICBjb25zb2xlLmxvZyh3cmFwcGVyLmZpbmQoQnV0dG9uQWN0aW9uKSk7XG4gIH0pO1xufSlcbiJdfQ==