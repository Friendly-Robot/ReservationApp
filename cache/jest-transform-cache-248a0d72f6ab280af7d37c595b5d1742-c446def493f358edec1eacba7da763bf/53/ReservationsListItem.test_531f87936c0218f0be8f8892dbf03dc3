1d897788e88762bb03756424c0223947
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

require("react-native");

var _react = _interopRequireDefault(require("react"));

var _ReservationsListItem = _interopRequireDefault(require("../ReservationsListItem"));

var _reactTestRenderer = _interopRequireDefault(require("react-test-renderer"));

describe('<ReservationsListItem />', function () {
  var props;
  beforeEach(function () {
    props = {
      reservations: [{
        __typename: 'Reservations',
        hotelName: 'test'
      }]
    };
  });
  it('should be defined', function () {
    expect(_ReservationsListItem.default).toBeDefined();
  });
  it('renders correctly', function () {
    var tree = _reactTestRenderer.default.create(_react.default.createElement(_ReservationsListItem.default, props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlJlc2VydmF0aW9uc0xpc3RJdGVtLnRlc3QuanMiXSwibmFtZXMiOlsiZGVzY3JpYmUiLCJwcm9wcyIsImJlZm9yZUVhY2giLCJyZXNlcnZhdGlvbnMiLCJfX3R5cGVuYW1lIiwiaG90ZWxOYW1lIiwiaXQiLCJleHBlY3QiLCJSZXNlcnZhdGlvbnNMaXN0SXRlbSIsInRvQmVEZWZpbmVkIiwidHJlZSIsInJlbmRlcmVyIiwiY3JlYXRlIiwidG9KU09OIiwidG9NYXRjaFNuYXBzaG90Il0sIm1hcHBpbmdzIjoiOztBQUlBOztBQUNBOztBQUNBOztBQUdBOztBQUVBQSxRQUFRLENBQUMsMEJBQUQsRUFBNkIsWUFBTTtBQUN6QyxNQUFJQyxLQUFKO0FBQ0FDLEVBQUFBLFVBQVUsQ0FBQyxZQUFNO0FBQ2ZELElBQUFBLEtBQUssR0FBRztBQUNORSxNQUFBQSxZQUFZLEVBQUUsQ0FBQztBQUFDQyxRQUFBQSxVQUFVLEVBQUUsY0FBYjtBQUE2QkMsUUFBQUEsU0FBUyxFQUFFO0FBQXhDLE9BQUQ7QUFEUixLQUFSO0FBR0QsR0FKUyxDQUFWO0FBTUFDLEVBQUFBLEVBQUUsQ0FBQyxtQkFBRCxFQUFzQixZQUFNO0FBQzVCQyxJQUFBQSxNQUFNLENBQUNDLDZCQUFELENBQU4sQ0FBNkJDLFdBQTdCO0FBQ0QsR0FGQyxDQUFGO0FBSUFILEVBQUFBLEVBQUUsQ0FBQyxtQkFBRCxFQUFzQixZQUFNO0FBQzVCLFFBQU1JLElBQUksR0FBR0MsMkJBQVNDLE1BQVQsQ0FBZ0IsNkJBQUMsNkJBQUQsRUFBMEJYLEtBQTFCLENBQWhCLEVBQXFEWSxNQUFyRCxFQUFiOztBQUNBTixJQUFBQSxNQUFNLENBQUNHLElBQUQsQ0FBTixDQUFhSSxlQUFiO0FBQ0QsR0FIQyxDQUFGO0FBSUQsQ0FoQk8sQ0FBUiIsInNvdXJjZXNDb250ZW50IjpbIi8qKlxuICogQGZvcm1hdFxuICovXG5cbmltcG9ydCAncmVhY3QtbmF0aXZlJztcbmltcG9ydCBSZWFjdCBmcm9tICdyZWFjdCc7XG5pbXBvcnQgUmVzZXJ2YXRpb25zTGlzdEl0ZW0gZnJvbSAnLi4vUmVzZXJ2YXRpb25zTGlzdEl0ZW0nO1xuXG4vLyBOb3RlOiB0ZXN0IHJlbmRlcmVyIG11c3QgYmUgcmVxdWlyZWQgYWZ0ZXIgcmVhY3QtbmF0aXZlLlxuaW1wb3J0IHJlbmRlcmVyIGZyb20gJ3JlYWN0LXRlc3QtcmVuZGVyZXInO1xuXG5kZXNjcmliZSgnPFJlc2VydmF0aW9uc0xpc3RJdGVtIC8+JywgKCkgPT4ge1xuICBsZXQgcHJvcHM7XG4gIGJlZm9yZUVhY2goKCkgPT4ge1xuICAgIHByb3BzID0ge1xuICAgICAgcmVzZXJ2YXRpb25zOiBbe19fdHlwZW5hbWU6ICdSZXNlcnZhdGlvbnMnLCBob3RlbE5hbWU6ICd0ZXN0J31dLFxuICAgIH07XG4gIH0pXG5cbiAgaXQoJ3Nob3VsZCBiZSBkZWZpbmVkJywgKCkgPT4ge1xuICAgIGV4cGVjdChSZXNlcnZhdGlvbnNMaXN0SXRlbSkudG9CZURlZmluZWQoKTtcbiAgfSk7XG5cbiAgaXQoJ3JlbmRlcnMgY29ycmVjdGx5JywgKCkgPT4ge1xuICAgIGNvbnN0IHRyZWUgPSByZW5kZXJlci5jcmVhdGUoPFJlc2VydmF0aW9uc0xpc3RJdGVtIHsuLi5wcm9wc30gLz4pLnRvSlNPTigpO1xuICAgIGV4cGVjdCh0cmVlKS50b01hdGNoU25hcHNob3QoKTtcbiAgfSk7XG59KVxuIl19