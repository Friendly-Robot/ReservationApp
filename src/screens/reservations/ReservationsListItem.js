/*
* @flow
*/

import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

import theme from '../../theme';

type ItemProps = {
  reservations: Array<Object>,
};

export default ({ reservations = [] }: ItemProps) => {
  if (reservations[0].__typename !== 'Reservation') return null;
  const hotelName = reservations[0].hotelName;
  return (
    <View style={styles.container}>
      <Text style={styles.hotelName}>{ hotelName }</Text>

      {reservations.map(res => (
        <View key={res.id} style={styles.itemContainer}>
          <Text style={styles.itemLabel}>Guest: <Text style={styles.itemName}>{res.name}</Text></Text>
          <View style={styles.itemDatesContainer}>
            <Text style={styles.itemDates}>
              {res.arrivalDate}
              <Text style={styles.hotelName}>{` -> `}</Text>
              {res.departureDate}
            </Text>
          </View>
        </View>
      ))}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: theme.COLOR_WHITE,
    borderColor: theme.COLOR_DARK,
    borderRadius: theme.BORDER_RADIUS_SMALL,
    borderWidth: theme.BORDER_WIDTH_MEDIUM,
    marginHorizontal: theme.BORDER_MARGIN_HORIZONTAL,
    marginVertical: theme.BORDER_MARGIN_VERTICAL,
    paddingHorizontal: theme.BORDER_PADDING_HORIZONTAL,
    paddingVertical: theme.BORDER_PADDING_VERTICAL,
  },
  hotelName: {
    color: theme.COLOR_DARK,
    fontSize: theme.FONT_SIZE_LARGE,
    fontWeight: theme.FONT_WEIGHT_HEAVY,
    marginBottom: theme.BORDER_MARGIN_VERTICAL,
  },
  itemContainer: {
    paddingVertical: theme.BORDER_PADDING_VERTICAL,
  },
  itemDates: {
    color: theme.COLOR_DARK,
    fontSize: theme.FONT_SIZE_SMALL,
  },
  itemDatesContainer: {
    flexDirection: 'row',
  },
  itemLabel: {
    color: theme.COLOR_DARK,
    fontSize: theme.FONT_SIZE_MEDIUM,
  },
  itemName: {
    color: theme.COLOR_PRIMARY,
    fontSize: theme.FONT_SIZE_MEDIUM,
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
  },
});
