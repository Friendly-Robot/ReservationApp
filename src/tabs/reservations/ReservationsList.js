/*
* @flow
*/

import React from 'react';
import { ScrollView, StyleSheet, View } from 'react-native';

import ReservationsHeader from './ReservationsHeader';
import ReservationsListItem from './ReservationsListItem';
import ButtonAction from '../../common/ButtonActionRound';

import theme from '../../theme';

type listProps = {
  hotels: Array<T>,
  setShowScheduler: Function,
};

export default ReservationsList = ({hotels = [], setShowScheduler = () => {}}: listProps) => (
  <View style={styles.container}>
    <ReservationsHeader title={'Reservations'} />
    <ButtonAction onPress={() => setShowScheduler(true)} />
    <ScrollView contentContainerStyle={styles.scrollContainer}>
      {hotels.map(reservations => {
        if (reservations.length === 0) return null;
        if (!reservations[0].hotelName) return null;
        return (
          <ReservationsListItem
            key={reservations[0].hotelName}
            reservations={reservations}
          />
        );
      })}
    </ScrollView>
  </View>
);

const styles = StyleSheet.create({
  container: {
    backgroundColor: theme.COLOR_WHITE,
    flex: 1,
  },
  scrollContainer: {
    flexGrow: 1,
    paddingBottom: 215, // offset marginTop
    marginTop: 215,
  }
});
