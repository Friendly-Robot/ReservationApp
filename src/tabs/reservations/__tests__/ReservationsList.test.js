/**
 * @format
 */

import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import jest from 'jest-mock';

import ReservationsList from '../ReservationsList';
import ReservationsListItem from '../ReservationsListItem';
import ButtonAction from '../../../common/ButtonActionRound';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('<ReservationsList />', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      hotels: [[{
        hotelName: 'test',
        __typename: 'Reservation',
        id: '123456',
        name: 'Andy',
        arrivalDate: '04/22/2020',
        departureDate: '04/22/9999',
      }]],
      setShowScheduler: jest.fn(),
    };

    wrapper = shallow(<ReservationsList {...props} />);
  });

  it('should be defined', () => {
    expect(ReservationsList).toBeDefined();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<ReservationsList />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders ReservationsListItem', () => {
    expect(wrapper.find(ReservationsListItem)).toBeDefined();
  })

  it('renders a ButtonAction to allow opening ReservationsScheduler', () => {
    expect(wrapper.find(ButtonAction)).toBeDefined();
  });
});
