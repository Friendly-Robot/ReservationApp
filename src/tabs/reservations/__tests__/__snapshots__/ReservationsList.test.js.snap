// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReservationsList /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(255, 255, 255)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(255, 255, 255)",
          "flexDirection": "row",
          "height": 65,
          "zIndex": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(30, 147, 242)",
            "fontSize": 16,
            "marginLeft": 15,
          }
        }
      >
        Reservations
      </Text>
    </View>
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "height": 150,
          "width": 750,
          "zIndex": 2,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/common/img/hotel-front-desk.jpeg",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 150,
              "width": 750,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "bottom": 25,
        "position": "absolute",
        "right": 15,
        "zIndex": 10,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(30, 147, 242)",
          "borderRadius": 1000,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "shadowColor": undefined,
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.5,
          "shadowRadius": 2,
          "width": 50,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(255, 255, 255)",
            "fontSize": 35,
          }
        }
      >
        +
      </Text>
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flexGrow": 1,
        "marginTop": 215,
        "paddingBottom": 215,
      }
    }
  >
    <View />
  </RCTScrollView>
</View>
`;
